src/tasks/tasks.controller.ts

```diff
@@ -37,4 +39,19 @@ export class TasksController {
       );
     }
   }
+
+  @Get(":id")
+  async findTaskById(@Param("id") id: number): Promise<Task | null> {
+    try {
+      return await this.tasksService.findTaskById(id);
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(
+        `Failed to find task with ID ${id}`,
+        error.message,
+      );
+    }
+  }
```

Comment:
```suggestion
      return this.tasksService.findTaskById(Id)
```

Should be fine since you're handling the errors in service layer and there is no need to throw Internal Server Errors explicitly